core/trino-spi/src/main/java/io/trino/spi/type/DateTimeEncoding.java

```diff
@@ -33,6 +33,15 @@ private static long pack(long millisUtc, short timeZoneKey)
         return (millisUtc << MILLIS_SHIFT) | (timeZoneKey & TIME_ZONE_MASK);
     }
 
+    public static long packMillisUtc(long millisUtc)
```

Comment:
What timezone does this use?

Reply:
In the existing parquet reader we were using `packDateTimeWithZone(epochMillis, UTC_KEY)`. Since `timeZoneKey` is 0 for `UTC_KEY` we created this simpler method without the bit masking in `pack(long millisUtc, short timeZoneKey)`

Reply:
We shouldn't assume 0 is UTC_KEY. Also, if a caller uses a constant in the call to `pack(xxx, UTC_KEY)`, when that gets inlined it should constant fold and the bit masking should be eliminated. Have you shown that to not be the case and for this to matter?

Also, the name of the method is confusing, since it's not clear whether it means "pack the given millis utc", or "pack the given millis utc in utc time zone"

Reply:
> We shouldn't assume 0 is UTC_KEY.

That is exactly what the current reader is doing

Reply:
I've removed this method and replaced it with `packDateTimeWithZone(epochMillis, UTC_KEY)` everywhere